### DISCUSSION TITLE
Are API v3 and v4 equivalent?

### DISCUSSION THREAD

**Discussion by KnowYourselves** [community/community#24559] [Apps, API and Webhooks]
<p>Hello!</p>
<p>I’m currently getting to know the Github API and I’ve been tasked with migrating an App that uses the API from v3 to v4. My problem is that there are certain requests from the v3 API that I can’t really migrate. For example, I don’t see a mutation in the v4 to delete a repository, or to add a collaborator to an existing repository.</p>
<p>¿Am I reading the documentation wrong, or there are v3 requests that as of yet, cannot be migrated to v4?</p>
<p>Since I’m learning GraphQl it may be me not reading the documentation properly. Thanks!</p>

**Comment by username7171** [comment]
<p>It depends on how do you mean by equivalent.</p>
<p>They could do the same thing in different ways.</p>
<p>V4 is just a better way since it could do something in one request that would be several requests if you do it in v3.</p>
<p>If you are trying to remove a repository, make sure your token has the scope  “delete_repo”  in order to do so. </p>

**Comment by KnowYourselves** [comment]
<p>Thanks for the answer!</p>
<p>In that case, how do you delete a repository using v4? Is not that it’s giving me an error, it’s that I didn’t find a mutation that would actually delete a repo, while in v3 there’s just an endpoint for that.</p>

**Comment by mkrakowitzer** [comment, marked as answer]
<p>I have found that V4 is missing a lot of functionality compared to v3. I have the same issues as you, I need to manage teams by deleting teams, removing members from teams and inviting members. None of these are possible with V4 from what I can find, only v3. </p>

**Comment by KnowYourselves** [comment]
<p>That’s the answer then, thanks!</p>
<p>What I’ll do at least is make a module were all requests are made and make and interface for it. Then I’ll update the V3 requests as V4 is updated. Thanks again!</p>
